pkg/controllers/localbuild/controller.go

```diff
@@ -384,8 +384,12 @@ func (r *LocalbuildReconciler) reconcileGitRepo(ctx context.Context, resource *v
 			Source: v1alpha1.GitRepositorySource{
 				Type: repoType,
 			},
-			GitURL:         resource.Status.Gitea.ExternalURL,
-			InternalGitURL: resource.Status.Gitea.InternalURL,
+			Provider: v1alpha1.Provider{
+				Name:             v1alpha1.GitProviderGitea,
```

Comment:
Given that we are hard-coding it to Gitea, what would be the step for users to choose GitHub instead? Are the users expecetd to add a separate provider to support Github in a way that it co-exists with Gitea? or is there a way to choose alternatives?

Reply:
No way to choose an alternative with this PR. To do it, we need to update the CLI and other controllers. Doing it in all one PR will result in a massive PR. Trying to keep a PR small.